rustls/src/client/tls13.rs

```diff
@@ -302,77 +505,312 @@ pub(super) fn initial_key_share(
     group.start()
 }
 
-/// This implements the horrifying TLS1.3 hack where PSK binders have a
-/// data dependency on the message they are contained within.
-pub(super) fn fill_in_psk_binder(
```

Comment:
Split the refactoring from the logic additions (a one-variant enum would be fine for this).